xds/src/test/java/io/grpc/xds/XdsLoadReportStoreTest.java

```diff
@@ -0,0 +1,261 @@
+            .getStreamTracerFactory()
+            .newClientStreamTracer(STREAM_INFO, new Metadata());
+    Duration interval = Duration.newBuilder().setNanos(342).build();
+    ClusterStats expectedLoadReport = buildClusterStats(interval,
+        Collections.singletonList(buildUpstreamLocalityStats(locality1, 0, 0, 1)));
+    assertClusterStatsEqual(expectedLoadReport, loadStore.generateLoadReport(interval));
+
+    // Make another load report should not reset count for calls in progress.
+    assertClusterStatsEqual(expectedLoadReport, loadStore.generateLoadReport(interval));
+
+    tracer.streamClosed(Status.OK);
+    expectedLoadReport = buildClusterStats(interval,
+        Collections.singletonList(buildUpstreamLocalityStats(locality1, 1, 0, 0)));
+    assertClusterStatsEqual(expectedLoadReport, loadStore.generateLoadReport(interval));
+
+    // Make another load report should reset finished calls count for calls finished.
+    expectedLoadReport = buildClusterStats(interval,
+        Collections.singletonList(buildUpstreamLocalityStats(locality1, 0, 0, 0)));
+    assertClusterStatsEqual(expectedLoadReport, loadStore.generateLoadReport(interval));
+
+    // PickResult within the same locality should aggregate to the same counter.
+    PickResult pickResult2 = PickResult.withSubchannel(fakeSubchannel);
+    PickResult interceptedPickResult2 = loadStore.interceptPickResult(pickResult2, locality1);
+    assertThat(localityLoadCounters).hasSize(1);
+    interceptedPickResult1
+        .getStreamTracerFactory()
+        .newClientStreamTracer(STREAM_INFO, new Metadata())
+        .streamClosed(Status.ABORTED);
+    interceptedPickResult2
+        .getStreamTracerFactory()
+        .newClientStreamTracer(STREAM_INFO, new Metadata())
+        .streamClosed(Status.CANCELLED);
+    expectedLoadReport = buildClusterStats(interval,
+        Collections.singletonList(buildUpstreamLocalityStats(locality1, 0, 2, 0)));
+    assertClusterStatsEqual(expectedLoadReport, loadStore.generateLoadReport(interval));
+
+    expectedLoadReport = buildClusterStats(interval,
+        Collections.singletonList(buildUpstreamLocalityStats(locality1, 0, 0, 0)));
+    assertClusterStatsEqual(expectedLoadReport, loadStore.generateLoadReport(interval));
+
+    Locality locality2 =
+        Locality.newBuilder()
+            .setRegion("test_region2")
+            .setZone("test_zone")
+            .setSubZone("test_subzone")
+            .build();
+    loadStore.addLocality(locality2);
+    PickResult pickResult3 = PickResult.withSubchannel(fakeSubchannel);
+    PickResult interceptedPickResult3 = loadStore.interceptPickResult(pickResult3, locality2);
+    assertThat(localityLoadCounters).containsKey(locality2);
+    assertThat(localityLoadCounters).hasSize(2);
+    interceptedPickResult3
+        .getStreamTracerFactory()
+        .newClientStreamTracer(STREAM_INFO, new Metadata());
+    List<UpstreamLocalityStats> upstreamLocalityStatsList = new ArrayList<>();
```

Comment:
You can use `Arrays.asList()` to save a few characters.

Reply:
ACK.